read.go

```diff
@@ -26,42 +26,38 @@ func (s *Segment) getDocStoredMetaAndUnCompressed(docNum uint64) (meta, data []b
 
 	// document chunk coder
 	chunkI := docNum / uint64(defaultDocumentChunkSize)
-	s.m.Lock()
-	var storedFieldDecompressed []byte
-	var ok bool
-	if storedFieldDecompressed, ok = s.decompressedStoredFieldChunks[chunkI]; !ok {
+	storedFieldDecompressed := s.decompressedStoredFieldChunks[chunkI]
```

Comment:
I don't think we ever actually succeed in updating items inside the map.  Although you mutate `storedFieldDecompressed` I believe this is a copy, not the item the map.  See this example I think is analagous: https://go.dev/play/p/s0UrS7nD_fg

Reply:
Yeah, you are right, i changed it. https://github.com/blugelabs/ice/pull/16/commits/34f03d613a41c74d3e77345cbc2921114a45a1d0